### PR TITLE
refine allocation cmake

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

Others
### PR changes

Others
### Describe

Refine allocation cmake, make all source files compiled into targets one static library (instead of several libraries) so all files  can be compiled parallelly.

It reduces about 3~4min compilation time.

Related work: #43247

before
image

after
image

### CODE REVIEW

**Review by XieYunshen — DISMISSED** [review]
LGTM 
代码移动触发RUN_TYPE的检查

**Review by wanghuancoder — DISMISSED** [review]
LGTM

**Review by From00 — DISMISSED** [review]
LGTM

**Review by sneaxiy — DISMISSED** [review]

File: paddle/fluid/memory/allocation/CMakeLists.txt [hunk 1]
```diff
@@ -1,264 +1,193 @@
...
-  cc_test(
-    thread_local_allocator_test
-    SRCS thread_local_allocator_test.cc
-    DEPS thread_local_allocator)
-  if(CUDA_VERSION GREATER_EQUAL 10.2)
-    nv_library(
-      cuda_virtual_mem_allocator
-      SRCS cuda_virtual_mem_allocator.cc
-      DEPS dynload_cuda)
+if(CUDA_VERSION GREATER_EQUAL 10.2)
```
Comment (tiancaishaonvjituizi) [inline paddle/fluid/memory/allocation/CMakeLists.txt]:
```suggestion
if(CUDA_VERSION VERSION_GREATER_EQUAL 10.2)
```
这样是不是更正确一些

File: paddle/fluid/memory/allocation/CMakeLists.txt [hunk 2]
```diff
@@ -1,264 +1,193 @@
...
+  system_allocator_test
+  SRCS system_allocator_test.cc
+  DEPS allocator)
+
+cc_test(
+  buddy_allocator_test
+  SRCS buddy_allocator_test.cc
+  DEPS allocator)
+
+function(file_download_and_uncompress URL NAME)
```
Comment (tiancaishaonvjituizi) [inline paddle/fluid/memory/allocation/CMakeLists.txt]: 建议这个函数改一个名字。从名字来看这是一个下载压缩包并解压的纯工具函数，但实际上它修改了 third_party_deps，只应该用来下载第三方库

File: paddle/fluid/memory/allocation/CMakeLists.txt (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/memory/allocation/CMakeLists.txt]: 是的，updated

File: paddle/fluid/memory/allocation/CMakeLists.txt (hunk 2 above)
Comment (zhiqiu) [reply paddle/fluid/memory/allocation/CMakeLists.txt]: 多谢，仔细看了下，这块的逻辑有个bug，我换了种实现。

**Review by XieYunshen — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**